hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -504,13 +504,14 @@ boolean checkInvocationTimeout() {
         long maxCallTimeout = invocationFuture.getMaxCallTimeout();
         long expirationTime = op.getInvocationTime() + maxCallTimeout;
 
+        boolean done = invocationFuture.isDone();
         boolean hasResponse = pendingResponse != null;
         boolean hasWaitingThreads = invocationFuture.getWaitingThreadsCount() > 0;
         boolean notExpired = maxCallTimeout == Long.MAX_VALUE
                 || expirationTime < 0
                 || expirationTime >= Clock.currentTimeMillis();
 
-        if (hasResponse || hasWaitingThreads || notExpired) {
+        if (hasResponse || hasWaitingThreads || notExpired || done) {
```

Comment:
So apparently we have invocations that have received a response (otherwise future.set would not have been called), but the invocation is not deregistered from the registry?